### PR TITLE
[Fixed] Compatibility with Ruby v3

### PR DESCRIPTION
**Pull request by manuelmhtr**
:question: What
---
- Fixed bug: `login` always throws a `Belvo::BelvoAPIError`.
- On `APISession#detail` method, add `**` operator to send `params` as keyword argument, rather than hash.
- Update dependencies.

:speech_balloon: Why
---
`detail` methods failed in Ruby v3 because `APISession#get` received `params` as hash, raising an `ArgumentError`.

### CODE REVIEW

**Review by Janee — DISMISSED** [review]
👍

**Review by benitocas88 — DISMISSED** [review]
After test the compatibility with `ruby` `2` and `3` works fine

File: Gemfile [hunk 1]
```diff
@@ -6,14 +6,4 @@ source 'https://rubygems.org'
...
-gem 'coveralls_reborn', require: false
-gem 'faraday'
-gem 'faraday_middleware'
-gem 'rake', '~> 12.0'
-gem 'rspec', '~> 3.0'
-gem 'rubocop', '~> 0.81.0', require: false
-gem 'rubocop-rspec', require: false
-gem 'typhoeus'
-gem 'danger', '8.2.1'
-gem 'webmock'
```
Comment (manuelmhtr) [inline Gemfile:18]: They are required in `gemspec`

File: Gemfile [hunk 2]
```diff
@@ -6,14 +6,4 @@ source 'https://rubygems.org'
...
 
-gem 'coveralls_reborn', require: false
-gem 'faraday'
-gem 'faraday_middleware'
-gem 'rake', '~> 12.0'
-gem 'rspec', '~> 3.0'
-gem 'rubocop', '~> 0.81.0', require: false
-gem 'rubocop-rspec', require: false
-gem 'typhoeus'
-gem 'danger', '8.2.1'
```
Comment (manuelmhtr) [inline Gemfile:17]: I removed `danger` apparently is not being used.

File: lib/belvo/http.rb [hunk 3]
```diff
@@ -52,7 +52,7 @@ def raise_for_status(response)
     #   to Belvo API else False.
     def authenticate
       @session.basic_auth(@key_id, @key_password)
-      response = @session.get('')
+      response = @session.get('api/')
```
Comment (manuelmhtr) [inline lib/belvo/http.rb:55]: This was causing problems since `@session` does not include `api/` by default.

**Comment by manuelmhtr** [comment]
@AlbertMaristany version `0.19.2` is not working (`login` always throws a `Belvo::BelvoAPIError`). I fixed it in this pull request.

**Review by Janee — DISMISSED** [review]
👍

**Comment by manuelmhtr** [comment]
@AlbertMaristany any thoughts?

**Review by AlbertMaristany — DISMISSED** [review]

File: lib/belvo/version.rb [hunk 4]
```diff
@@ -2,5 +2,5 @@
 
 module Belvo
   # belvo-ruby current version
-  VERSION = '0.19.2'
+  VERSION = '0.19.3'
```
Comment (AlbertMaristany) [inline lib/belvo/version.rb:5]: Thanks for the PR. Could you leave the version as `0.19.2`? Belvo team will manage the versioning when releasing a new version. Thanks!

File: Gemfile (hunk 2 above)
Comment (AlbertMaristany) [reply Gemfile:17]: We have a danger workflow in place which is failing right now.

File: lib/belvo/version.rb (hunk 4 above)
Comment (manuelmhtr) [reply lib/belvo/version.rb]: No problem. Done 👍

File: Gemfile (hunk 2 above)
Comment (AlbertMaristany) [reply Gemfile:17]: Screenshot from 2021-12-24 09-22-46

We need danger back.

Comment (manuelmhtr) [reply Gemfile:17]: Sorry about that, It's fixed now.

File: lib/belvo/http.rb [hunk 5]
```diff
@@ -41,7 +41,7 @@ def raise_for_status(response)
         raise RequestError.new(
           'Request error',
           response.status,
-          response.body.to_s
+          response.body
```
Comment (manuelmhtr) [inline lib/belvo/http.rb:44]: Detail was difficult to use as a hash string, sending now as a hash.

Comment (Janee) [reply lib/belvo/http.rb:44]: Awesome!

**Review by Janee — APPROVED** [review]

**Review by alex-belvo — APPROVED** [review]

File: lib/belvo/http.rb (hunk 5 above)
Comment (alex-belvo) [reply lib/belvo/http.rb:44]: This change is breaking all responses.

**Comment by manuelmhtr** [comment]
Closing this PR due to lack of open source culture :/

**Comment by luisguerrerolab** [comment]
Hey, @manuelmhtr:

I apologize for not letting you know that, due to internal protocols, we had to open a new PR (#113) where we could effectively merge your contribution.

Fortunately we already released a fix to our public Ruby gem containing this fix:
https://rubygems.org/gems/belvo/versions/0.19.3

Thank you so much for raising this issue and helping us to improve our Ruby SDK.

**Pull request closed without merging**